psutil/_psutil_sunos.c

```diff
@@ -139,12 +190,37 @@ psutil_proc_name_and_args(PyObject *self, PyObject *args) {
     py_name = PyUnicode_DecodeFSDefault(info.pr_fname);
     if (!py_name)
         goto error;
-    py_args = PyUnicode_DecodeFSDefault(info.pr_psargs);
+
+    /* SunOS truncates arguments to length PRARGSZ, very likely args are truncated.
+     * The only way to retrieve full command line is to parse process memory */
+    if (info.pr_argc && strlen(info.pr_psargs) == PRARGSZ-1) {
+        argv = psutil_read_raw_args(info, procfs_path, &argc);
+        if (argv) {
+            argv_plain = cstrings_array_to_string(argv, argc, ' ');
+            if (argv_plain) {
+                py_args = PyUnicode_DecodeFSDefault(argv_plain);
+                free(argv_plain);
+            } else {
+                PyErr_NoMemory();
```

Comment:
You should `PyErr_NoMemory` from `cstrings_array_to_string`, not here

Reply:
Well, I can. But isn't it better to leave the utility function clean? Right now it returns NULL only in one case (let's say like strcat), and I handle this case from function which works with python objects.
If you don't like I will change this of course

Reply:
The rationale is that if the utility function changes and need to raise something different than `PyErr_NoMemory` then the caller has no way to know what to do